src/App.php

```diff
@@ -1623,7 +1618,394 @@ public function checkURL()
+					info(Core\L10n::t("You must be logged in to use addons. "));
+				} else {
+					include_once "addon/{$this->module}/{$this->module}.php";
+					if (function_exists($this->module . '_module')) {
+						LegacyModule::setModuleFile("addon/{$this->module}/{$this->module}.php");
+						$this->module_class = 'Friendica\\LegacyModule';
+						$this->module_loaded = true;
+					}
+				}
+			}
+
+			// Controller class routing
+			if (! $this->module_loaded && class_exists('Friendica\\Module\\' . ucfirst($this->module))) {
+				$this->module_class = 'Friendica\\Module\\' . ucfirst($this->module);
+				$this->module_loaded = true;
+			}
+
+			/* If not, next look for a 'standard' program module in the 'mod' directory
+			 * We emulate a Module class through the LegacyModule class
+			 */
+			if (! $this->module_loaded && file_exists("mod/{$this->module}.php")) {
+				LegacyModule::setModuleFile("mod/{$this->module}.php");
+				$this->module_class = 'Friendica\\LegacyModule';
+				$this->module_loaded = true;
+			}
+
+			/* The URL provided does not resolve to a valid module.
+			 *
+			 * On Dreamhost sites, quite often things go wrong for no apparent reason and they send us to '/internal_error.html'.
+			 * We don't like doing this, but as it occasionally accounts for 10-20% or more of all site traffic -
+			 * we are going to trap this and redirect back to the requested page. As long as you don't have a critical error on your page
+			 * this will often succeed and eventually do the right thing.
+			 *
+			 * Otherwise we are going to emit a 404 not found.
+			 */
+			if (! $this->module_loaded) {
+				// Stupid browser tried to pre-fetch our Javascript img template. Don't log the event or return anything - just quietly exit.
+				if (!empty($_SERVER['QUERY_STRING']) && preg_match('/{[0-9]}/', $_SERVER['QUERY_STRING']) !== 0) {
+					exit();
+				}
+
+				if (!empty($_SERVER['QUERY_STRING']) && ($_SERVER['QUERY_STRING'] === 'q=internal_error.html') && isset($dreamhost_error_hack)) {
+					logger('index.php: dreamhost_error_hack invoked. Original URI =' . $_SERVER['REQUEST_URI']);
+					goaway($this->getBaseURL() . $_SERVER['REQUEST_URI']);
+				}
+
+				logger('index.php: page not found: ' . $_SERVER['REQUEST_URI'] . ' ADDRESS: ' . $_SERVER['REMOTE_ADDR'] . ' QUERY: ' . $_SERVER['QUERY_STRING'], LOGGER_DEBUG);
+
+				header($_SERVER["SERVER_PROTOCOL"] . ' 404 ' . Core\L10n::t('Not Found'));
+				$tpl = get_markup_template("404.tpl");
+				$this->page['content'] = replace_macros($tpl, [
+					'$message' =>  Core\L10n::t('Page not found.')
+				]);
+			}
+		}
+
+		/**
+		 * Load current theme info
+		 */
+		$theme_info_file = 'view/theme/' . $this->getCurrentTheme() . '/theme.php';
+		if (file_exists($theme_info_file)) {
+			require_once $theme_info_file;
+		}
+
+
+		/* initialise content region */
+
+		if ($this->getMode()->isNormal()) {
+			Core\Addon::callHooks('page_content_top', $this->page['content']);
+		}
+
+		/**
```

Comment:
`/**` initiates the doxygen documentation please use `/*` or `/* *` for comments